frontend/src/Project/service/loadMachine.ts

```diff
@@ -111,15 +144,47 @@ async function getLabelRasters(source: TiffPixelSource) {
   return channels;
 }
 
-type Raster = { data: Uint8Array | Int32Array; width: number; height: number };
+function reshapeRaw(channels: TypedArray[][][], min: number, max: number) {
+  // Normalize each pixel to 0-255 for rendering
+  for (let c = 0; c < channels.length; c++) {
+    for (let z = 0; z < channels[c].length; z++) {
+      for (let y = 0; y < channels[c][z].length; y++) {
+        for (let x = 0; x < channels[c][z][y].length; x++) {
```

Comment:
Is it really necessary to iterate over x and y too? Can you not apply the rescaling to an entire frame?

Reply:
You can apply rescaling to an entire frame using forEach but it seems like these sorts of callback functions are actually slower than regular for loops. I will do some benchmarking and try to use gpu.js to speed it up

Reply:
> Can you look into why the coverage decreased so dramatically? I don't mind small drops but a 64% drop seems to indicate a problem in how the tests are running.

I reran the backend tests and the coverage appears normal now; I'm not sure what happened on the initial run.